### PR TITLE
Add DOAS modeling capability

### PR DESCRIPTION
**Pull request by nealkruis**
Adds capabilities described in #148.

### CODE REVIEW

File: src/CGCOMP.CPP [hunk 1]
```diff
@@ -834,6 +834,163 @@ void ANDAT::ad_DupHERV( const ANDAT& ad)		// dup HERV values
...
+			{	
+				if (izx->iz_IsDOAS() && izx->iz_doas == ss)
+				{
+					if (izx->iz_vfMin > 0)
+					{
+						izx->iz_vfMin *= supFlowRatio;
+					}
+				}	
+			}
+		}
```
Comment (nealkruis) [inline src/CGCOMP.CPP:901]: This can maybe be turned into a function to eliminate repetition on the Exhaust side.

File: src/CGCOMP.CPP [hunk 2]
```diff
@@ -834,6 +834,163 @@ void ANDAT::ad_DupHERV( const ANDAT& ad)		// dup HERV values
...
+	oa_supAF.af_amf = 0.;
+	oa_exhAF.af_amf = 0.;
+	IZXRAT* izx;
+	RLUP( IzxR, izx)
+	{	
+		if (izx->iz_IsDOAS() && izx->iz_doas == ss)
+		{
+			if (izx->iz_vfMin > 0)
+			{
+				oa_supAF.af_amf += izx->iz_vfMin * Top.tp_rhoMoistOSh / 60.;
```
Comment (nealkruis) [inline src/CGCOMP.CPP:952]: Note: Air mass flow for energy calculations is based on outdoor air density, but the actual air mass flows at the zones are based on supply air density.

File: src/CGCOMP.CPP [hunk 3]
```diff
@@ -1441,6 +1618,15 @@ void IZXRAT::iz_GetExteriorConditions(
 		iz_cpr * windPresV	// pressure coeff * velocity pressure
 		- iz_hz * iz_rho2;	// plus stack
 }	// IZXRAT::iz_GetExteriorConditions
+//-----------------------------------------------------------------------------
+void IZXRAT::iz_GetDOASConditions()
+// set source conditions to DOAS supply
+{
+	DOAS* pDOAS = doasR.GetAt(iz_doas);
+	iz_air2 = static_cast<AIRSTATE>(pDOAS->oa_supAF);
+	iz_rho2 = iz_air2.as_RhoMoist();
```
Comment (nealkruis) [inline src/CGCOMP.CPP:1627]: Following on previous comment: The density used for the Air Net is at the supply air state, where the density used for fan energy calculations is outdoor air density.

**Comment by nealkruis** [comment]
@chipbarnaby, in addition to the comments above, it appears that `ad_tdFan` could be used uninitialized for non-fan IZXFERs.

**Review by chipbarnaby — APPROVED** [review]

**Pull request merged**